framework/db/src/com/cloud/utils/db/Merovingian2.java

```diff
@@ -63,9 +63,7 @@
     private Merovingian2(long msId) {
         super(MerovingianMBean.class, false);
         _msId = msId;
-        Connection conn = null;
-        try {
-            conn = TransactionLegacy.getStandaloneConnectionWithException();
+        try (Connection conn = TransactionLegacy.getStandaloneConnectionWithException();) {
```

Comment:
yes, you are right. it needs to be addressed though. I think the close should then happen in the catch instead of automatically.